### PR TITLE
Bryan attempt targetsdkversion29

### PR DESCRIPTION
**Pull request by bryanwussow**
I attempted some changes to go to TargetSdkVersion 29
Task :app:compileDebugJavaWithJavac is running into this:
```
BuildListener#buildStarted(Gradle) has been deprecated. This is scheduled to be removed in Gradle 7.0.
	at build_a78a6cw5jlqihit19vysxenql.run(C:\Users\BWussow\Documents\GIT local\HearThisAndroid\app\build.gradle:1)
	(Run with --stacktrace to get the full stack trace of this deprecation warning.)
```
but I can't locate BuildListener anywhere.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/hearthisandroid/3)

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
 ##############################################################################
 ##
 ##  Gradle start up script for UN*X
```
Comment (JohnThomson) [inline gradlew:21]: This doesn't appear changed but surprises me. AFAIK HTA has never been compiled on a Unix/Linux system. I've always built it with Android Studio on Windows. So maybe this file isn't used at all? Or maybe Android Studio has a way to run Unix scripts?

File: app/build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
 android {
-    compileSdkVersion 27
+    compileSdkVersion 29
 
     defaultConfig {
         applicationId "org.sil.hearthis"
         minSdkVersion 18 // gradle insists it can't be smaller than this
-        targetSdkVersion 27
+        targetSdkVersion 29
```
Comment (JohnThomson) [inline app/build.gradle:9]: This is the really essential change (the compileSdk change is very likely required to match). Were the others done by some automated process?

File: app/build.gradle [hunk 3]
```diff
@@ -23,18 +23,18 @@ android {
 }
 
 dependencies {
-    testCompile 'junit:junit:4.12'
+    testImplementation 'junit:junit:4.12'
```
Comment (JohnThomson) [inline app/build.gradle:26]: I'm not familiar with this, but guessing it's a language change. Is this the result of using a new version of Gradle? Unless something demands it, I would not attempt to deal with switching Gradle versions at the same time as changing SDK version.

**Comment by JohnThomson** [comment]
Bryan, most of the changes here are to a file I'm not even familiar with...I think the original version must have been generated by some wizard that created the original project. I've added a few comments inline.
Not sure how familiar you are with Android programming and Android Studio. In case you're not aware of this, to build for SDK 29 you will have to download that SDK. I would do this using a tool called SDK Manager. I can never remember which icon launches it, but ctrl-shift-A brings up a search-for-command dialog in which you can find it by typing sdk.
The command "Sync project with Gradle files" (which you can find similarly) is often useful in solving mysterious build problems.
BuildListener sounds like something in the development environment. I'm not aware of such a class in HTA.
It's possible if you only recently installed Android Studio that you automatically got a later version of Gradle than HTA was built with. That might be a source of problems (some of which you may already have discovered and fixed.)
Sorry I can't be of more help. Andrew Polk might have ideas...he's been involved in some of the recent SDK upgrades for Bloom Reader.

**Comment by andrew-polk** [comment]
I spent a little time seeing if I could get this to work. I submitted my own branch (called TargetSdk29) in the main fork. It is slightly different than Bryan's branch.
I didn't really test, but it does compile and ran on my 4.4 test device.
It doesn't pass the lint process, however. So while `assemble` works, `build` doesn't. This is due to using an old library.

**Comment by andrew-polk** [comment]
Superceded by #4

**Pull request closed without merging**